### PR TITLE
perf-tuning: Add document for topn-limit-push-down

### PR DESCRIPTION
**Pull request by LittleFall**
### What is changed, added or deleted? (Required)

Add Chinese document for topn-limit-push-down.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### CODE REVIEW

File: reference/performance/topn-push-down.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+title: 执行计划缓存
```
Comment (kissmydb) [inline reference/performance/topn-push-down.md:2]: 是否写错了？

**Comment by LittleFall** [comment]
PTAL @winoros  @kissmydb

File: topn-limit-push-down.md [hunk 2]
```diff
@@ -3,4 +3,110 @@ title: TopN 和 Limit 下推
 category: performance
 ---
 
-# TopN 和 Limit 下推
\ No newline at end of file
+# TopN 和 Limit 下推
+
+sql 中的 limit 子句在 tidb 查询计划树中对应 limit 算子节点，order by 子句在查询计划树中对应 sort 算子节点，此外，我们会将相邻的 limit 和 sort 组合成 topn 算子节点，表示按某个排序规则提取记录的前 n 项。从另一方面来说，limit 节点等价于一个排序规则为空的 topn 节点。
```
Comment (zz-jason) [inline topn-limit-push-down.md:8]:
```suggestion
SQL 中的 LIMIT 子句在 TiDB 查询计划树中对应 Limit 算子节点，ORDER BY 子句在查询计划树中对应 Sort 算子节点，此外，我们会将相邻的 Limit 和 Sort 算子组合成 TopN 算子节点，表示按某个排序规则提取记录的前 N 项。从另一方面来说，Limit 节点等价于一个排序规则为空的 TopN 节点。
```

File: topn-limit-push-down.md [hunk 3]
```diff
@@ -3,4 +3,110 @@ title: TopN 和 Limit 下推
...
+
+和谓词下推类似，topn（及 limit，下同）下推将查询计划树中的 topn 计算尽可能下推到距离数据源最近的地方，以尽早完成数据的过滤，进而显著地减少数据传输或计算的开销。
+
+如果要关闭这个规则，可以在参照[优化规则及表达式下推的黑名单](/blacklist-control-plan.md)中的关闭方法。
+
+## 示例
+
+以下通过一些例子对 topn 下推进行说明。
+
+### 示例1：下推到存储层
```
Comment (zz-jason) [inline topn-limit-push-down.md:18]:
```suggestion
### 示例 1：下推到存储层 Coprocessor
```

File: topn-limit-push-down.md [hunk 4]
```diff
@@ -3,4 +3,110 @@ title: TopN 和 Limit 下推
 category: performance
 ---
 
-# TopN 和 Limit 下推
\ No newline at end of file
+# TopN 和 Limit 下推
+
+SQL 中的 LIMIT 子句在 TiDB 查询计划树中对应 Limit 算子节点，ORDER BY 子句在查询计划树中对应 Sort 算子节点，此外，我们会将相邻的 Limit 和 Sort 算子组合成 TopN 算子节点，表示按某个排序规则提取记录的前 N 项。从另一方面来说，Limit 节点等价于一个排序规则为空的 TopN 节点。
+
+和谓词下推类似，topn（及 limit，下同）下推将查询计划树中的 topn 计算尽可能下推到距离数据源最近的地方，以尽早完成数据的过滤，进而显著地减少数据传输或计算的开销。
```
Comment (zz-jason) [inline topn-limit-push-down.md:10]: 这里的大小写也改下吧

File: topn-limit-push-down.md (hunk 3 above)
Comment (zz-jason) [reply topn-limit-push-down.md:18]: 注意中英文之间的空格，其他标题也一起修改下吧

**Comment by winoros** [comment]
@kissmydb PTAL thx

File: topn-limit-push-down.md [hunk 5]
```diff
@@ -3,4 +3,122 @@ title: TopN 和 Limit 下推
...
+|   │ └─TableReader_26             | 10.00    | root      |               | data:TopN_25                                    |
+|   │   └─TopN_25                  | 10.00    | cop[tikv] |               | test.t.a, offset:0, count:10                    |
+|   │     └─TableFullScan_24       | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo                  |
+|   └─TableReader_30(Probe)        | 10000.00 | root      |               | data:TableFullScan_29                           |
+|     └─TableFullScan_29           | 10000.00 | cop[tikv] | table:s       | keep order:false, stats:pseudo                  |
++----------------------------------+----------+-----------+---------------+-------------------------------------------------+
+8 rows in set (0.01 sec)
+```
+
+在该查询中，TopN 算子的排序规则仅依赖于外表 t 中的列，可以将 TopN 下推到了 Join 之前进行一次计算，以减少 Join 时的计算开销。除此之外，同样将 TopN 下推到了存储层中。
```
Comment (yikeke) [inline topn-limit-push-down.md:67]:
```suggestion
在该查询中，TopN 算子的排序规则仅依赖于外表 t 中的列，可以将 TopN 下推到 Join 之前进行一次计算，以减少 Join 时的计算开销。除此之外，TiDB 同样将 TopN 下推到了存储层中。
```
这里补一下主语？

**Pull request merged**